Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -315,311 +545,498 @@ void CGContextSetFillColorSpace(CGContextRef pContext, CGColorSpaceRef colorSpac
 /**
  @Status Interoperable
 */
-void CGContextSelectFont(CGContextRef ctx, const char* name, CGFloat size, CGTextEncoding encoding) {
-    ctx->Backing()->CGContextSelectFont((char*)name, size, encoding);
+void CGContextSetGrayFillColor(CGContextRef context, CGFloat gray, CGFloat alpha) {
+    CGContextSetRGBFillColor(context, gray, gray, gray, alpha);
 }
 
 /**
  @Status Interoperable
 */
-CGPoint CGContextGetTextPosition(CGContextRef ctx) {
-    CGPoint ret;
-    ctx->Backing()->CGContextGetTextPosition(&ret);
+void CGContextSetRGBFillColor(CGContextRef context, CGFloat r, CGFloat g, CGFloat b, CGFloat a) {
+    ComPtr<ID2D1SolidColorBrush> brush;
+    FAIL_FAST_IF_FAILED(context->RenderTarget()->CreateSolidColorBrush({ r, g, b, a }, &brush));
+    FAIL_FAST_IF_FAILED(brush.As(&context->CurrentGState().fillBrush));
+}
 
-    return ret;
+/**
+ @Status Caveat
+ @Notes Manually converts CMYK to RGB, and does not involve the colorspace.
+*/
+void CGContextSetCMYKFillColor(CGContextRef context, CGFloat cyan, CGFloat magenta, CGFloat yellow, CGFloat black, CGFloat alpha) {
+    CGContextSetRGBFillColor(context,
+                             (1.0f - cyan) * (1.0f - black),
+                             (1.0f - magenta) * (1.0f - black),
+                             (1.0f - yellow) * (1.0f - black),
+                             alpha);
 }
+#pragma endregion
 
+#pragma region Graphics States
 /**
  @Status Interoperable
 */
-void CGContextSaveGState(CGContextRef ctx) {
-    ctx->Backing()->CGContextSaveGState();
+void CGContextSaveGState(CGContextRef context) {
+    auto& oldState = context->CurrentGState();
```

Comment:
This needs to be evaluated case-by-case; I'd like to keep the control flow simple and not put too much of the logic in Impl; my ideal mental model for these things is that they are dumb data containers that we manipulate externally (ala the CF model)

---
In reply to: [82920043](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r82920043) [](ancestors = 82920043)